lib/features/docker_save.js

```diff
@@ -23,26 +27,81 @@ export default class DockerSave {
     let imgStream = await image.get();
     let zipStream = zlib.createGzip();
     imgStream.pipe(zipStream).pipe(fs.createWriteStream(pathname));
-    await waitForEvent(zipStream, 'end');
+    await new Promise((accept, reject) => {
+      zipStream.on('end', accept);
+      zipStream.on('error', (error) => reject(error));
+    });
     debug('tar written');
 
     let stat = await fs.stat(pathname);
     debug(stat.size);
     let uploadStream = fs.createReadStream(pathname);
-    let expiration = new Date(Date.now() + task.runtime.dockerSave.expiration);
+    let expiration = taskcluster.fromNow(task.runtime.dockerSave.expiration);
 
-    await uploadToS3(task, uploadStream, ARTIFACT_NAME, expiration, {
+    await uploadToS3(task, uploadStream, 'public/dockerImage.tar.gz', expiration, {
```

Comment:
As discussed before please strip `.gz` we don't need it when `content-encoding: gzip` is set...